test/csuite/wt2403_lsm_workload/main.c

```diff
@@ -227,7 +228,8 @@ main(int argc, char *argv[])
     testutil_check(session->open_cursor(session, name, NULL, "overwrite", &wcursor));
     for (i = 0; i < NUM_DOCS; i++) {
         wcursor->set_key(wcursor, i);
-        testutil_check(wcursor->remove(wcursor));
+        if (wcursor->search(wcursor) == 0)
+            testutil_check(wcursor->remove(wcursor));
```

Comment:
We should ignore `WT_NOTFOUND`? I'm not sure we care, but this is 2x tree searches.

Reply:
Yes, I will change it to something similar to what you suggested above.

Reply:
@mm-ng, I made the obvious change here and the test program breaks. I'm suspicious there's something more subtle going on with LSM. 
```
diff --git a/test/csuite/wt2403_lsm_workload/main.c b/test/csuite/wt2403_lsm_workload/main.c
index 7555c6d3f..cde8ef465 100644
--- a/test/csuite/wt2403_lsm_workload/main.c
+++ b/test/csuite/wt2403_lsm_workload/main.c
@@ -109,7 +109,6 @@ main(int argc, char *argv[])
     WT_SESSION *session, *session2;
     pthread_t thread;
     uint64_t i;
-    int ret;
 
     char str[] = "0000000000000000";
 
@@ -177,11 +176,10 @@ main(int argc, char *argv[])
     testutil_check(session2->open_cursor(session2, name, NULL, "next_random=true", &rcursor));
 
     /* Delete all but one document */
-    testutil_check(session->open_cursor(session, name, NULL, "overwrite", &wcursor));
+    testutil_check(session->open_cursor(session, name, NULL, NULL, &wcursor));
     for (i = 0; i < NUM_DOCS - 1; i++) {
         wcursor->set_key(wcursor, i);
-        if (wcursor->search(wcursor) == 0)
-            testutil_check(wcursor->remove(wcursor));
+        testutil_check_error_ok(wcursor->remove(wcursor), WT_NOTFOUND);
     }
     testutil_check(wcursor->close(wcursor));
     printf("%d documents deleted\n", NUM_DOCS - 1);
@@ -226,14 +224,10 @@ main(int argc, char *argv[])
     testutil_check(pthread_join(thread, NULL));
 
     /* Delete everything. Check for infinite loops */
-    testutil_check(session->open_cursor(session, name, NULL, "overwrite", &wcursor));
+    testutil_check(session->open_cursor(session, name, NULL, NULL, &wcursor));
     for (i = 0; i < NUM_DOCS; i++) {
         wcursor->set_key(wcursor, i);
-        if ((ret = wcursor->search(wcursor)) == WT_NOTFOUND)
-            fprintf(
-              stderr, "cursor.remove: %s\n", wcursor->session->strerror(wcursor->session, ret));
-        else
-            testutil_check(wcursor->remove(wcursor));
+        testutil_check_error_ok(wcursor->remove(wcursor), WT_NOTFOUND);
     }
     testutil_check(wcursor->close(wcursor));
```